routes/project/location/kubernetes_cluster.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class Clover
+  hash_branch(:project_location_prefix, "kubernetes-cluster") do |r|
+    r.get api? do
+      kubernetes_cluster_list
+    end
+
+    r.on NAME_OR_UBID do |kc_name, kc_ubid|
+      if kc_name
+        r.post api? do
+          post_kubernetes_cluster(kc_name)
+        end
+
+        filter = {Sequel[:kubernetes_cluster][:name] => kc_name}
+      else
+        filter = {Sequel[:kubernetes_cluster][:id] => UBID.to_uuid(kc_ubid)}
+      end
+
+      filter[:location] = @location
+      kc = @project.kubernetes_clusters_dataset.first(filter)
+
+      next (r.delete? ? 204 : 404) unless kc
+
+      r.get true do
+        authorize("KubernetesCluster:view", kc.id)
+        response.headers["Cache-Control"] = "no-store"
```

Comment:
We added this header only to the PostgreSQL detail page because it contains a confidential connection string. If you don't have anything like that, there's no need to add it.